marimo/_runtime/watch/watch.py

```diff
@@ -0,0 +1,341 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+import os
+import sys
+import threading
+import time
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Callable, TypeVar, cast
+
+from marimo._output.rich_help import mddoc
+from marimo._runtime.context import (
+    ContextNotInitializedError,
+    get_context,
+    runtime_context_installed,
+)
+from marimo._runtime.side_effect import SideEffect
+from marimo._runtime.state import State
+from marimo._runtime.threads import Thread
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+
+T = TypeVar("T")
+
+WATCHER_SLEEP_INTERVAL = 1.0
+
+# For testing only - do not use in production
+_TEST_SLEEP_INTERVAL: float | None = None
+
+
+def write_side_effect(data: str | bytes) -> None:
```

Comment:
Since you are exporting the entire `watch` module, the private APIs should be underscored

Reply:
`__init__` only exposes `file` and `directory` in `__all__`

But I can break up this file into `path`, `file`, `directory` for some code cleanliness

Reply:
the test `tests/test_api.py::test_api` should let you know. 

i don't think `__all__` actually enforces anything

Reply:
Nice. Glad we have that test